File: back end/rcf-soluciones-informaticas/products-service/src/main/java/com/rcf/productsservice/service/DiscountTypeService.java
package com.rcf.productsservice.service;

import com.rcf.productsservice.dto.DiscountTypeRequest;
import com.rcf.productsservice.dto.DiscountTypeResponse;

import java.util.List;

public interface DiscountTypeService {
    List<DiscountTypeResponse> getAllDiscountTypes();
    DiscountTypeResponse getDiscountTypeById(Long id);
    DiscountTypeResponse saveDiscountType(DiscountTypeRequest discountTypeRequest);
    DiscountTypeResponse updateDiscountType(Long id, DiscountTypeRequest discountTypeRequest);
    void deleteDiscountType(Long id);
}

File: front end/rcf-soluciones-informaticas-angular/src/app/shared/services/product-services/discount-type-service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import DiscountType from '../../model/DiscountType';
import DiscountTypeRequest from '../../model/Products-Service/DiscountTypeRequest.model';

@Injectable({
  providedIn: 'root'
})
export class DiscountTypeService {
  private apiUrl = 'http://localhost:8080/products'; 
  private domain = 'discount-types';

  constructor(private http: HttpClient) {
  }

  getDiscountTypes() : Observable<DiscountType[]> {
    return this.http.get<DiscountType[]>(`${this.apiUrl}/${this.domain}`);
  }
  getDiscountTypeById(id: number) : Observable<DiscountType> {
    return this.http.get<DiscountType>(`${this.apiUrl}/${this.domain}/${id}`);
  }
  createDiscountType(discountType: DiscountTypeRequest) : Observable<DiscountType> {
    return this.http.post<DiscountType>(`${this.apiUrl}/${this.domain}`, discountType);
  }
  updateDiscountType(id: number, discountType: DiscountTypeRequest) : Observable<DiscountType> {
    return this.http.put<DiscountType>(`${this.apiUrl}/${this.domain}/${id}`, discountType);
  }
  deleteDiscountType(id: number) : Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${this.domain}/${id}`);
  }


}
